src/kubernetes_cluster/spec/controller/controller_runtime.rs

```diff
@@ -51,33 +57,42 @@ pub open spec fn continue_reconcile<K: ResourceView, R: Reconciler<K>>() -> Cont
                 &&& !R::reconcile_error(s.ongoing_reconciles[cr_key].local_state)
                 // Split cases:
                 // (1) there is a pending request which is destined for kubernetes api;
-                // (3) there is no pending request which is destined for kubernetes api and there is a external response;
-                // (4) there is no pending request or external response;
+                // (2) there is a pending external api request (so that we should send a message to the external api) and an external api output
```

Comment:
Do you mean "a pending external api input"?
If so you can change it to "so that we should feed the input to the external api"